File: public/dsa/arrays/max_consecutive_ones/optimal_2/max_consecutive_ones.cpp

#include <bits/stdc++.h>
using namespace std;

int getMaxConsecutiveOnes(vector<int> &A) {
    int max_count = 0;
    int cur_count = 0;
    for (int v : A) {
        if (v == 1) {
            cur_count++;
            max_count = max(max_count, cur_count);
        } else {
            cur_count = 0;
        }
    }
    return max_count;
}

// tests-start
int main() {
    vector<int> arr = {1, 1, 3, 2, 3, 1, 1, 1};
    cout << getMaxConsecutiveOnes(arr) << endl;
    return 0;
}
// tests-end
/*output
3
*/

File: public/dsa/arrays/max_consecutive_ones/optimal_2/max_consecutive_ones.py
from typing import List


class Solution:
    def getMaxConsecutiveOnes(self, arr: List[int]) -> int:
        max_count = 0
        cur_count = 0
        for v in arr:
            if v == 1:
                cur_count += 1
                max_count = max(cur_count, max_count)
            else:
                cur_count = 0
        return max_count


# tests-start
arr = [1, 1, 3, 2, 3, 1, 1, 1]
sol = Solution()
print(sol.getMaxConsecutiveOnes(arr))
# tests-end

'''output
3
'''
